src/services/theme/emotion.tsx

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { FunctionComponent, PropsWithChildren } from 'react';
+import { ThemeProvider } from '@emotion/react';
+
+import { useEuiTheme } from './hooks';
+
+/**
+ * @see https://emotion.sh/docs/theming
+ */
+export const EuiEmotionThemeProvider: FunctionComponent<
+  PropsWithChildren<{}>
+> = ({ children }) => {
+  const euiThemeContext = useEuiTheme();
+  return <ThemeProvider theme={euiThemeContext}>{children}</ThemeProvider>;
```

Comment:
I'm also pretty 50/50 on providing the entire theme context (i.e. `colorMode` and `modify` alongside with the `euiTheme` vars) or just `euiTheme` itself, i.e.

```suggestion
  const { euiTheme } = useEuiTheme();
  return <ThemeProvider theme={euiTheme}>{children}</ThemeProvider>;
```

To be honest, I would assume most consumers won't need color mode info, but I think it might be syntactically confusing to return a `theme` prop that doesn't match the return structure of our own `useEuiTheme()` hook. 🤷